### ISSUE TITLE
Remove testharness.js test-level timeouts (often used for async_test)

### ISSUE THREAD

**Issue by foolip** [web-platform-tests/wpt#11120]
See discussion at https://github.com/w3c/web-platform-tests/issues/11029#issuecomment-389499369

https://chromium-review.googlesource.com/c/chromium/src/+/1061753 shows what tests currently use this timeout.

Steps to this issue:
 * Remove the use of `{ timeout: 1234 }` from all tests
 * Run https://chromium-review.googlesource.com/c/chromium/src/+/1061753 through Chromium's CQ again (or Gecko's)
 * Remove support for the timeout entirely.

@jgraham, plan SGTY?

**Comment by foolip** [comment]
What would remain of the `Test` constructors second argument `properties` is the "assert" member, AFAICT only used in the output. @jgraham, should we want to get rid of that as well?

**Comment by jgraham** [comment]
Yeah, I'm happy with all of that plan, including removing the `assert` support.

If you make a PR it should automatically run all those tests through the gecko CI.

**Comment by foolip** [comment]
I've sent https://chromium-review.googlesource.com/c/chromium/src/+/1070155 to see if anything uses that.

**Comment by foolip** [comment]
It looks like these strings show up in their own column in the report. Just having the information in the test name or assertion message would work. Needing to pick one means this will require some human judgement to do.

**Comment by qiuzhong** [comment]
I'm interested in working on this issue. 

@foolip, I didn't find any strings like `{timeout: 1234}` in the html tests above. How did you get this list?

**Comment by foolip** [comment]
@qiuzhong, that's great! Try `git grep -P "timeout:\s*\d+"` in the root of the repository, that matches 223 files, most of which look like this: `async_test("getCurrentPosition success callback tests", {timeout:20000})`. The first step is to update those tests to look like this instead: `async_test("getCurrentPosition success callback tests")`.

**Comment by foolip** [comment]
It would be a good idea to split those changes into multiple PRs since otherwise a lot of reviewers will get pulled in. Maybe one PR for each of html, IndexedDB, websockets, mediacapture-streams and cookies, and then one for everything else?

Once we think all uses are gone, I'll run a change similar to https://chromium-review.googlesource.com/1070155 through Chromium's CQ to see if anything was missed. Then we can just delete timeouts.

**Comment by qiuzhong** [comment]
@foolip, thanks for your good advice. It's really a good idea to split this update into several PRs. 

I tried the `git grep -P "timeout:\s*\d+"` command and the output list differ from the one you provided above.  Anyway, I'll update these test cases in serveral PRs. Now I'll list the submodules of these tests:
- [ ] IndexedDB
- [ ] cookies
- [ ] cors
- [ ] css
- [ ] custom-elements
- [ ] encrypted-media
- [ ] eventsource
- [ ] geolocation-API
- [ ] html
- [ ] mediacapture-streams
- [ ] old-tests
- [ ] requestidlecallback
- [ ] resource-timing
- [ ] resources
- [ ] shadow-dom
- [ ] tools
- [ ] web-nfc
- [ ] webauthn
- [ ] webrtc
- [ ] websockets
- [ ] xhr

**Comment by foolip** [comment]
@qiuzhong when you create the PRs, can you link to this issue so that's it easy to find them all? Thanks!

**Comment by qiuzhong** [comment]
@foolip,  There're promise_tests used in `cookies` tests and they have the `{timeout: 1234}` parameter as well. Should I remove them all?

https://github.com/web-platform-tests/wpt/blob/706ee77bda06b58f4e317bc317bc0e7c25d34f12/cookies/http-state/attribute-tests.html#L21

https://github.com/web-platform-tests/wpt/blob/706ee77bda06b58f4e317bc317bc0e7c25d34f12/cookies/http-state/attribute-tests.html#L54-L56

**Comment by foolip** [comment]
@FHorschig wrote that test. Given the use of an explicit timeout, I suspect that if we remove the test-level timeouts, then if any test doesn't finish, then the whole test would fail when wptrunner stops waiting for it. (This is different from testharness.js internally reaching a timeout and telling wptrunner.)

Can you start by checking what the effect on the test is of removing both of the lines you quoted?

**Comment by foolip** [comment]
@qiuzhong looks like there aren't too many left now, yay :)

Some of the directories marked as done in https://github.com/web-platform-tests/wpt/issues/11120#issuecomment-438511094 still have tests using timeouts on master though, so best to grep again.

**Comment by qiuzhong** [comment]
@foolip , thanks for reminding the missing async_test. I should double check after commit. See new PR #14227 for removing the rest `timeout` in async_test.

**Comment by qiuzhong** [comment]
Grepping the `timeout` in the root directory, there'are still many matches. But they're mainly in `setup` and `promise_test`. I have no idea about the impact of removing `timeout` in these statements.

**Comment by foolip** [comment]
The `promise_test` ones should be removed, the ones in `setup` should not.

**Comment by foolip** [comment]
(At least not to resolve this issue, perhaps they should separately, later.)

**Comment by foolip** [comment]
I've sent https://chromium-review.googlesource.com/c/chromium/src/+/1061753 to see what tests still need updating.

**Comment by foolip** [comment]
That job has finished and these are the tests that still need updating:
```
cookies/http-state/attribute-tests.html
cookies/http-state/charset-tests.html
cookies/http-state/chromium-tests.html
cookies/http-state/comma-tests.html
cookies/http-state/domain-tests.html
cookies/http-state/general-tests.html
cookies/http-state/mozilla-tests.html
cookies/http-state/name-tests.html
cookies/http-state/ordering-tests.html
cookies/http-state/path-tests.html
cookies/http-state/value-tests.html
css/css-transitions/before-DOMContentLoaded-001.html
css/css-transitions/before-load-001.html
css/css-transitions/changing-while-transition.html
css/css-transitions/detached-container-001.html
css/css-transitions/hidden-container-001.html
css/css-transitions/properties-value-002.html
css/css-transitions/properties-value-003.html
css/css-transitions/properties-value-auto-001.html
css/css-transitions/properties-value-implicit-001.html
css/css-transitions/properties-value-inherit-003.html
css/css-transitions/pseudo-elements-001.html
geolocation-API/getCurrentPosition_IDL.https.html
webrtc/simplecall.https.html
webrtc/simplecall-no-ssrcs.https.html
xhr/abort-during-upload.htm
xhr/send-redirect-bogus.htm
```

@qiuzhong do you have PRs out for all of those?

**Comment by foolip** [comment]
Looks like https://github.com/web-platform-tests/wpt/pull/14253 covers some, yay :)

**Comment by foolip** [comment]
@qiuzhong with your two most recent PRs merged, I think the tests in css/css-transitions/ are all that remain.

**Comment by qiuzhong** [comment]
Besides the css/css-transitions/, there are other tests in css/css-regions/ that need to remove the `timeout`. The `timeout` is passed as an argument so they escaped from grepping `timeout:\s*\d+`.

**Comment by qiuzhong** [comment]
@foolip , the tests you listed above should be all covered now.

**Comment by qiuzhong** [comment]
@foolip, you listed the steps to remove testharness.js test-level timeouts:

> * Remove the use of { timeout: 1234 } from all tests
> * Run https://chromium-review.googlesource.com/c/chromium/src/+/1061753 through Chromium's CQ again (or Gecko's)
> * Remove support for the timeout entirely.

I think the first step was done. But I don't know the status of step 2 so I'm not sure if it's time to start step 3.

**Comment by foolip** [comment]
I've started another run of https://chromium-review.googlesource.com/c/chromium/src/+/1061753, but please go ahead with step 3 in parallel, we can check the status of that before merging the final PR.

**Comment by foolip** [comment]
https://chromium-review.googlesource.com/c/chromium/src/+/1061753 is all green now, so good to go.

**Comment by foolip** [comment]
Resolved by https://github.com/web-platform-tests/wpt/pull/14309. Thanks @qiuzhong!
